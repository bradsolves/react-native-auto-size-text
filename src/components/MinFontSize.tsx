import * as React from 'react'
import { NativeSyntheticEvent, Text, TextLayoutEventData } from 'react-native'

import { AutoSizeTextProps } from '../index'

const MinFontSize = (props: AutoSizeTextProps) => {
  const { fontSize, children, style, numberOfLines, minFontSize } = props

  const [currentFont, setCurrentFont] = React.useState(fontSize)
  const handleTextMode = (e: NativeSyntheticEvent<TextLayoutEventData>) => {
    const { lines } = e.nativeEvent
    if (lines.length > (numberOfLines as number)) {
      setCurrentFont((currentFont as number) - 1)
    }
  }

  return (
    <Text
      numberOfLines={numberOfLines}
      style={[
        style,
        {
          fontSize: minFontSize,
        },
      ]}
      onTextLayout={(e) => {
        handleTextMode(e)
      }}
    >
      {children}
    </Text>
  )
}

export default MinFontSize
