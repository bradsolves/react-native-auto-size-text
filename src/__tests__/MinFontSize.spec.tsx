import { cleanup, render, waitFor } from '@testing-library/react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import MinFontSize from '../components/MinFontSize';

afterEach(cleanup);

describe('MinFontSize snapshot', () => {
  it('MinFontSize renders correctly', () => {
    const tree = renderer
      .create(
        <MinFontSize minFontSize={12} fontSize={14} numberOfLines={2}>
          Lorem impsum
        </MinFontSize>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('MinFontSize mode', () => {
  it('should pass the right props to Text component inside MinFontSize', async () => {
    const mockContent = 'Lorem ipsum';
    const mockNumberOfLines = 3;
    const mockFontSize = 24;
    const mockColor = 'red';

    const { getByTestId } = render(
      <MinFontSize
        style={{ color: mockColor }}
        numberOfLines={mockNumberOfLines}
        minFontSize={mockFontSize}
        fontSize={mockFontSize}
      >
        {mockContent}
      </MinFontSize>
    );

    const textComponent = getByTestId('min-font-size');
    const content = textComponent.props.children;
    const numberOfLines = textComponent.props.numberOfLines;
    const adjustsFontSizeToFit = textComponent.props.adjustsFontSizeToFit;
    const fontSize = textComponent.props.style[1].fontSize;
    const color = textComponent.props.style[0].color;

    await waitFor(() => {
      expect(content).toBe(mockContent);
      expect(numberOfLines).toBe(mockNumberOfLines);
      expect(fontSize).toBe(mockFontSize);
      expect(color).toBe(mockColor);
      expect(adjustsFontSizeToFit).toBeFalsy();
    });
  });
});
