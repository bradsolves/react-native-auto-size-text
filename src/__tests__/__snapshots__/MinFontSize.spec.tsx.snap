// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MinFontSize snapshot MinFontSize renders correctly 1`] = `
<Text
  onTextLayout={[Function]}
  style={
    Array [
      undefined,
      Object {
        "fontSize": 14,
      },
    ]
  }
  testID="min-font-size"
>
  Lorem impsum
</Text>
`;
